File: utils/testData.ts
/**
 * Centralised test data — update credentials / options here
 * without touching individual test files.
 */

export const VALID_USER = {
  username: 'tomsmith',
  password: 'SuperSecretPassword!',
} as const;

export const INVALID_USER = {
  username: 'wronguser',
  password: 'wrongpassword',
} as const;

export const DROPDOWN_OPTIONS = {
  placeholder: 'Please select an option',
  option1: 'Option 1',
  option2: 'Option 2',
} as const;

export const MESSAGES = {
  loginSuccess: 'You logged into a secure area!',
  loginFailure: 'Your username is invalid!',
  logoutSuccess: 'You logged out of the secure area!',
} as const;

File: selenium-java-pom/src/test/java/com/ailearning/seleniumpom/data/TestData.java
package com.ailearning.seleniumpom.data;

public final class TestData {
    public static final Credentials VALID_USER = new Credentials("tomsmith", "SuperSecretPassword!");
    public static final Credentials INVALID_USER = new Credentials("wronguser", "wrongpassword");

    public static final String DROPDOWN_PLACEHOLDER = "Please select an option";
    public static final String DROPDOWN_OPTION_1 = "Option 1";
    public static final String DROPDOWN_OPTION_2 = "Option 2";

    public static final String LOGIN_SUCCESS_MESSAGE = "You logged into a secure area!";
    public static final String LOGIN_FAILURE_MESSAGE = "Your username is invalid!";
    public static final String LOGOUT_SUCCESS_MESSAGE = "You logged out of the secure area!";

    private TestData() {
    }

    public record Credentials(String username, String password) {
    }
}
